src/analytics.js

```diff
@@ -98,35 +98,50 @@ const getSearchParams = () =>
 		} ),
 		{}
 	);
-const getAttributes = ( extra = {} ) =>
-	Object.entries( {
-		session: getSessionID(),
-		pageSession: pageSession,
-		url: window.location.origin + window.location.pathname,
-		host: window.location.hostname,
-		search: window.location.search,
-		hash: window.location.hash,
-		referer: document.referrer,
-		...getSearchParams(),
-		...utm,
-		...( Data.Attributes || {} ),
-		...extra,
-		...( _attributes || {} ),
-	} ).reduce( ( carry, [ name, value ] ) => ( {
-		...carry,
-		[ name ]: ( typeof value === 'function' ? value() : value ).toString(),
-	} ), {} );
-const getMetrics = ( extra = {} ) =>
-	Object.entries( {
-		elapsed: elapsed + ( Date.now() - start ),
-		scrollDepthMax,
-		scrollDepthNow,
-		...extra,
-		...( _metrics || {} ),
-	} ).reduce( ( carry, [ name, value ] ) => ( {
-		...carry,
-		[ name ]: Number( typeof value === 'function' ? value() : value ),
-	} ), {} );
+const prepareData = async ( value, sanitiseCallback ) => {
+	if ( typeof value === 'function' ) {
+		value = await value();
+	}
+	if ( ! Array.isArray( value ) ) {
+		value = [ value ];
+	}
+	return value.map( val => sanitiseCallback( val ) );
```

Comment:
Redundant arrow function, and also should use American English.

```suggestion
	return value.map( sanitizeCallback );
```

Reply:
Deerp